lib/google_search_data_viewer/keywords/google_search_client.ex

```diff
@@ -0,0 +1,15 @@
+defmodule GoogleSearchDataViewer.Keywords.GoogleSearchClient do
+  @base_url "https://www.google.com/search?q="
+  @headers [
+    {"User-Agent",
+     "Mozilla/5.0 (Macintosh; Intel Mac OS X 12_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"}
+  ]
+
+  def get_html(keyword) do
+    search_url = @base_url <> URI.encode(keyword)
+
+    {:ok, response} = HTTPoison.get(search_url, @headers)
```

Comment:
IMO, this function can be updated in two options:
1. If the error raising is unexpected in this function because the HTTP requests function may raise a pattern matching error. So we need error handling here, maybe return `error` atom with status.
2. If the error raising is expected and is a spec of this function, you can prefix the function name with `!` (get_html!)

Reply:
I initially returned an `error` status if the request failed but had issues trying to test this result naturally, ie I think I would need to use `Process.sleep` or mock it with a 500 server error?

Reply:
@liamstevens111 or you can try to create a VCR cassette that returns status code 500, then we can handle the error response from `HTTPoison`. The reason is when the app interacts with other 3rd services (ie: google) then we always need to handle the failed case.

Reply:
I attempted to use `stub` and `custom/manual` cassettes but I don't think this helps me as this just allows me to test the response. 

I need to test the error response match from `HTTPoison.get` and not the response details itself.